lib/install/deps.js

```diff
@@ -283,10 +283,12 @@ function computeVersionSpec (tree, child) {
   validate('OO', arguments)
   var requested
   var childReq = child.package._requested
-  if (childReq && (isNotEmpty(childReq.saveSpec) || (isNotEmpty(childReq.rawSpec) && isNotEmpty(childReq.fetchSpec)))) {
+  if (child.isLink) {
+    requested = npa.resolve(child.package.name, 'file:' + child.realpath, getTop(tree).path)
+  } else if (childReq && (isNotEmpty(childReq.saveSpec) || (isNotEmpty(childReq.rawSpec) && isNotEmpty(childReq.fetchSpec)))) {
     requested = child.package._requested
   } else if (child.package._from) {
-    requested = npa(child.package._from)
+    requested = npa(child.package._from, tree.path)
```

Comment:
Aren't all `_from` supposed to use `getTop(tree)`?

Reply:
`_from` is taken from what was in the `package.json` of the requiring package and is relative to that package, not to the top of the tree. When we save froms to the package-lock they're relative to the tree root.